Header header
float64 residual